src/luarocks/fetch.lua

```diff
@@ -244,21 +271,46 @@ function fetch.load_local_rockspec(filename, quick)
                       or rockspec.source.module
                       or ((filebase:match("%.lua$") or filebase:match("%.c$")) and ".")
                       or base
-   if rockspec.dependencies then
-      for i = 1, #rockspec.dependencies do
-         local parsed, err = deps.parse_dep(rockspec.dependencies[i])
-         if not parsed then
-            return nil, "Parse error processing dependency '"..rockspec.dependencies[i].."': "..tostring(err)
-         end
-         rockspec.dependencies[i] = parsed
-      end
-   else
-      rockspec.dependencies = {}
+   local ok, err = parse_deps(rockspec, "build_dependencies")
```

Comment:
I understand you're passing the table name as a string so you can do rockspec[field] = {} in line 188 above, but I'd just initialize both tables here instead and simplify the function above.
```lua
rockspec.dependencies = rockspec.dependencies or {}
local ok, err = parse_deps(rockspec.dependencies)
-- etc.
```
Minor thing, but it makes it clearer when reading that `parse_deps` isn't modifying the rest of the rockspec.